tools/fetch-retry.mjs

```diff
@@ -0,0 +1,41 @@
+// We are seeing quite a bit of 'fetch failed' cases in Github Actions that don't really reproduce
+// locally. We are likely hitting some limits there when attempting to parallelize. They are not consistent
+// so instead of reducing parallelism, we add a retry with backoff here.
+
+const originalFetch = globalThis.fetch
+
+const NUM_RETRIES = 5
+
+globalThis.fetch = async (...args) => {
+  let backoff = 100
+  for (let attempt = 1; attempt <= NUM_RETRIES; attempt++) {
+    try {
+      return await originalFetch.apply(globalThis, args)
+    } catch (error) {
+      let shouldRetry = false
+      // not ideal, but there is no error code for that
+      if (error.message === 'fetch failed' && attempt < NUM_RETRIES) {
```

Comment:
this is happening very randomly and does not seem related to any specific fetch target:
 - it might be happening in fetch that CLI is doing to fetch extensions and result in confusing error about site not being linked (even tho we specifically pass site_id to cli command in our e2e tests)
- it might be happening in data fetches in fixtures (both when preparing fixtures for integration tests and when building and deploying for e2e tests)

it seems more related to runners and hitting potential limits there